command/vsm_stats.go

```diff
@@ -409,3 +292,135 @@ func GetVolumeStats(address string, obj interface{}) (error, int) {
+		AvgWriteBlockCountPS int64
+	)
+
+	ReadIOPSi, _ := strconv.ParseInt(stats1.ReadIOPS, 10, 64)
+	ReadIOPSf, _ := strconv.ParseInt(stats2.ReadIOPS, 10, 64)
+	ReadIOPSPS := ReadIOPSf - ReadIOPSi
+
+	ReadTimePSi, _ := strconv.ParseInt(stats1.TotalReadTime, 10, 64)
+	ReadTimePSf, _ := strconv.ParseInt(stats2.TotalReadTime, 10, 64)
+	ReadTimePS := ReadTimePSf - ReadTimePSi
+
+	ReadBlockCountPSi, _ := strconv.ParseInt(stats1.TotalReadBlockCount, 10, 64)
+	ReadBlockCountPSf, _ := strconv.ParseInt(stats2.TotalReadBlockCount, 10, 64)
+	ReadBlockCountPS := ReadBlockCountPSf - ReadBlockCountPSi
+
+	RThroughput := ReadBlockCountPS
+	if ReadIOPSPS != 0 {
+		ReadLatency = ReadTimePS / ReadIOPSPS
+		AvgReadBlockCountPS = ReadBlockCountPS / ReadIOPSPS
+	} else {
+		ReadLatency = 0
+		AvgReadBlockCountPS = 0
+	}
+
+	WriteIOPSi, _ := strconv.ParseInt(stats1.WriteIOPS, 10, 64)
+	WriteIOPSf, _ := strconv.ParseInt(stats2.WriteIOPS, 10, 64)
+	WriteIOPSPS := WriteIOPSf - WriteIOPSi
+
+	WriteTimePSi, _ := strconv.ParseInt(stats1.TotalWriteTime, 10, 64)
+	WriteTimePSf, _ := strconv.ParseInt(stats2.TotalWriteTime, 10, 64)
+	WriteTimePS := WriteTimePSf - WriteTimePSi
+
+	WriteBlockCountPSi, _ := strconv.ParseInt(stats1.TotalWriteBlockCount, 10, 64)
+	WriteBlockCountPSf, _ := strconv.ParseInt(stats2.TotalWriteBlockCount, 10, 64)
+	WriteBlockCountPS := WriteBlockCountPSf - WriteBlockCountPSi
+
+	WThroughput := WriteBlockCountPS
+	if WriteIOPSPS != 0 {
+		WriteLatency = WriteTimePS / WriteIOPSPS
+		AvgWriteBlockCountPS = WriteBlockCountPS / WriteIOPSPS
+	} else {
+		WriteLatency = 0
+		AvgWriteBlockCountPS = 0
+	}
+
+	SectorSizePS, _ := strconv.ParseFloat(stats2.SectorSize, 64)
+	//	fmt.Println("hi", stats1.SectorSize, stats2.SectorSize)
+	SectorSizePS = SectorSizePS / 1048576
+	LogicalSize, _ := strconv.ParseFloat(stats2.UsedBlocks, 64)
+	LogicalSizePS := LogicalSize * SectorSizePS
+
+	ActualUsed, _ := strconv.ParseFloat(stats2.UsedLogicalBlocks, 64)
+	ActualUsedPS := ActualUsed * SectorSizePS
+
+	fmt.Println("------------------------------------")
+	// json formatting and showing default output
+	if c.Json == "json" {
+
+		stat1 := StatsArr{
+
+			IQN:    annotations.Iqn,
+			Volume: args[0],
+			Portal: annotations.TargetPortal,
+			Size:   annotations.VolSize,
+
+			ReadIOPS:  ReadIOPSPS,
+			WriteIOPS: WriteIOPSPS,
+
+			ReadThroughput:  float64(RThroughput) / 104857,
+			WriteThroughput: float64(WThroughput) / 104857,
+
+			ReadLatency:  float64(ReadLatency) / 1000000,
+			WriteLatency: float64(WriteLatency) / 1000000,
+
+			AvgReadBlockSize:  AvgReadBlockCountPS / 1024,
+			AvgWriteBlockSize: AvgWriteBlockCountPS / 1024,
+
+			SectorSize:  SectorSizePS,
+			ActualUsed:  ActualUsedPS,
+			LogicalSize: LogicalSizePS,
+		}
+
+		data, err := json.Marshal(stat1)
+
+		if err != nil {
+
+			panic(err)
+		}
+
+		os.Stdout.Write(data)
+
+		fmt.Println("\n------------------------------------")
+
+	} else {
+		fmt.Printf("%7s: %-48s\n", "IQN", annotations.Iqn)
```

Comment:
These are magic numbers as well. Please use typed constants or structures to derive meaning & hence maintainability.